test/widgets/actions_test.dart

```diff
@@ -278,19 +201,139 @@ void main() {
             'flag': 'read',
           });
     });
+  });
 
-    testWidgets('catch-all api errors', (WidgetTester tester) async {
+  group('updateMessageFlagsStartingFromAnchor', () {
+    String onCompletedMessage(int count) => 'onCompletedMessage($count)';
+    const progressMessage = 'progressMessage';
+    const onFailedTitle = 'onFailedTitle';
+
+    testWidgets('smoke test', (tester) async {
+      final narrow = TopicNarrow.ofMessage(eg.streamMessage());
+      final apiNarrow = narrow.apiEncode()..add(ApiNarrowIsUnread());
+      await prepare(tester);
+      connection.prepare(json: UpdateMessageFlagsForNarrowResult(
+        processedCount: 11, updatedCount: 3,
+        firstProcessedId: 1, lastProcessedId: 1980,
+        foundOldest: true, foundNewest: true).toJson());
+      final didPass = updateMessageFlagsStartingFromAnchor(
+        context: context, apiNarrow: apiNarrow, op: UpdateMessageFlagsOp.add, flag: MessageFlag.read,
+        includeAnchor: false, startingAnchor: AnchorCode.oldest, onCompletedMessage: onCompletedMessage,
+        onFailedTitle: onFailedTitle, progressMessage: progressMessage);
```

Comment:
nit: lines too long; and the arguments grouped on each line don't have anything thematically in common, so the wrapping together onto the same line makes it harder to spot any given parameter or see what's going on.

Instead, expanding out onto a line per argument is fine.

To reduce repetition between these test cases, can make a local helper that calls this function and provides all the values that don't change from one case to another.